### PR TITLE
[Feature](function) Support function quarters_add/sub for nereids

### PR DESCRIPTION
**Pull request by zclllyybb**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

1. BE already has its implementation. now add nereids function signatures.

```sql
mysql> select quarters_add("2020-12-12", 1);
+-----------------------------------------------+
| quarters_add(cast('2020-12-12' as DATEV2), 1) |
+-----------------------------------------------+
| 2021-03-12                                    |
+-----------------------------------------------+
1 row in set (0.10 sec)

mysql> select date_sub("2020-12-12", interval 10 quarter);
+------------------------------------------------+
| quarters_sub(cast('2020-12-12' as DATEV2), 10) |
+------------------------------------------------+
| 2018-06-12                                     |
+------------------------------------------------+
1 row in set (0.11 sec)
```

2. for date operations' template implementations in BE, we choose years_add to cover its standard testcases to test the base template.

### Release note

Support function quarters_add/sub

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [x] Yes. https://github.com/apache/doris-website/pull/1654

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by hello-stephen** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
wait https://github.com/apache/doris/pull/45265 merged. then add some testcase

**Comment by zclllyybb** [comment]
run buildall

**Review by zhangstar333 — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/rewrite/FEFunctions.java [hunk 1]
```diff
@@ -158,6 +158,30 @@ public static DateLiteral yearsAddDateTimeV2(LiteralExpr date, LiteralExpr year)
         return dateLiteral.plusYears((int) year.getLongValue());
     }
 
+    @FEFunction(name = "quarters_add", argTypes = { "DATETIME", "INT" }, returnType = "DATETIME")
+    public static DateLiteral quartersAdd(LiteralExpr date, LiteralExpr year) throws AnalysisException {
```
Comment (zhangstar333) [inline fe/fe-core/src/main/java/org/apache/doris/rewrite/FEFunctions.java:162]: year name seems some strange

Comment (zclllyybb) [reply fe/fe-core/src/main/java/org/apache/doris/rewrite/FEFunctions.java:162]: u r right, will fix~

**Comment by zclllyybb** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**